recipes/amaranth/meta.yaml

```diff
@@ -0,0 +1,53 @@
+{% set name = "amaranth" %}
+{% set version = "0.3" %}
+{% set sha256 = "102ca5e6b80c9c379e355c5e1c57263cba7be1c9cd0a58046568b427c1c977a1" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: v{{ version }}.tar.gz
+  url: https://github.com/amaranth-lang/{{ name }}/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+  patches:
+    - patches/0001-fix-extras.patch
+
+build:
+  number: 0
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vvv
+  script_env:
+    - SETUPTOOLS_SCM_PRETEND_VERSION={{ version }}
+  entry_points:
+    - amaranth-rpc = amaranth.rpc:main
+    - nmigen-rpc = nmigen.rpc:main
+requirements:
+  host:
+    - python >=3.7
+    - pip
+    - setuptools
+    - setuptools_scm
+  run:
+    - python >=3.7
+    - importlib_resources
```

Comment:
Upstream also has:
```suggestion
    - importlib_metadata
```

Reply:
I don't think this is still needed 3.8+?

Reply:
But the recipe says `python >=3.7` 😉

Reply:
conda-forge is only building 3.8+ these days though, im cool with either, I always forget I can't use selectors: https://github.com/conda-forge/staged-recipes/pull/22322/commits/27286641b46c6fe1ce84eb92020668b93eacf820

Reply:
Well, it's noarch so it will only build one package that relies on the version ranges to exclude it from environments. Right now, this recipe is installable on Python 3.7, but would need `importlib_metadata` to avoid an error..

Reply:
ok I can add